app/src/main/res/layout/fragment_planet_detail.xml

```diff
@@ -8,49 +8,103 @@
     android:background="@drawable/galaxy"
     tools:context=".ui.fragments.PlanetDetailFragment">
 
-    <ImageView
-        android:id="@+id/imagePlanet"
-        android:layout_width="250dp"
-        android:layout_height="250dp"
-        android:layout_marginTop="100dp"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent"
-        tools:src="@drawable/mars"/>
-
-    <TextView
-        android:id="@+id/textPlanetName"
-        android:layout_width="0dp"
-        android:layout_height="wrap_content"
-        android:layout_marginStart="16dp"
-        android:layout_marginTop="16dp"
-        android:layout_marginEnd="16dp"
-        android:fontFamily="sans-serif-black"
-        android:textAlignment="center"
-        android:textAllCaps="true"
-        android:textAppearance="@style/TextAppearance.AppCompat.Display2"
-        android:textColor="#FFFFFF"
-        android:textStyle="bold"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/imagePlanet"
-        tools:text="Mars" />
-
-    <TextView
-        android:id="@+id/textPlanetShortDesc"
-        android:layout_width="0dp"
-        android:layout_height="wrap_content"
-        android:layout_marginTop="8dp"
-        android:fontFamily="sans-serif"
-        android:textAlignment="center"
-        android:textAppearance="@style/TextAppearance.AppCompat.Body1"
-        android:textColor="#FFFFFF"
-        android:textSize="18sp"
-        android:textStyle="bold"
-        app:layout_constraintEnd_toEndOf="@+id/textPlanetName"
-        app:layout_constraintStart_toStartOf="@+id/textPlanetName"
-        app:layout_constraintTop_toBottomOf="@+id/textPlanetName"
-        tools:text="Where the androids live" />
+    <ScrollView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="vertical">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent">
```

Comment:
Made this so that my inner constraint layout is scrollable. Not sure this is the best way of doing it as well.

Reply:
why do you need the LinearLayout?

Reply:
try changing the line 22: to match_parent and adding android:fillViewport="true" to the scrollview, and remove the LinearLayout